src/python/pants/backend/jvm/tasks/coverage/scoverage.py

```diff
@@ -0,0 +1,152 @@
+from pants.util.dirutil import relativize_paths, safe_mkdir, safe_mkdir_for, safe_walk, touch
+
+
+report_generator = 'src/scala/org/pantsbuild/scoverage/report:gen2'
+
+
+class Scoverage(CoverageEngine):
+  """Class to run coverage tests with scoverage"""
+
+
+  class Factory(Subsystem):
+    options_scope = 'scoverage-runtime'
+
+    @classmethod
+    def create(cls, settings, targets, execute_java_for_targets):
+      """
+      :param settings: Generic code coverage settings.
+      :type settings: :class:`CodeCoverageSettings`
+      :param list targets: A list of targets to instrument and record code coverage for.
+      :param execute_java_for_targets: A function that accepts a list of targets whose JVM platform
+                                       constraints are used to pick a JVM `Distribution`. The
+                                       function should also accept `*args` and `**kwargs` compatible
+                                       with the remaining parameters accepted by
+                                       `pants.java.util.execute_java`.
+      """
+
+      return Scoverage(settings, targets, execute_java_for_targets)
+
+
+  @staticmethod
+  def register_junit_options(register, register_jvm_tool):
+
+    def scoverage_runtime_jar(**kwargs):
+      return JarDependency(org='com.twitter.scoverage', name='scalac-scoverage-runtime_2.12',
+        rev='1.0.1-twitter', **kwargs)
+
+    register_jvm_tool(register,
+      'scalac-scoverage-runtime',
+      classpath=[
+        scoverage_runtime_jar()
+      ])
+
+  def __init__(self, settings, targets, execute_java_for_targets):
+    """
+    :param settings: Generic code coverage settings.
+    :type settings: :class:`CodeCoverageSettings`
+    :param list targets: A list of targets to instrument and record code coverage for.
+    :param execute_java_for_targets: A function that accepts a list of targets whose JVM platform
+                                     constraints are used to pick a JVM `Distribution`. The function
+                                     should also accept `*args` and `**kwargs` compatible with the
+                                     remaining parameters accepted by
+                                     `pants.java.util.execute_java`.
+    """
+    self._settings = settings
+    self._context = settings.context
+    self._targets = targets
+    self._target_filters = []    # TODO(sameera): add target filtering for scoverage
+    self._execute_java = functools.partial(execute_java_for_targets, targets)
+
+  def _iter_datafiles(self, output_dir):
+    for root, _, files in safe_walk(output_dir):
+      for f in files:
+        if f.startswith("scoverage"):
+          yield os.path.join(root, f)
+
+  def _iter_datadirs(self, output_dir):
+    for root, dirs, _ in safe_walk(output_dir):
+      for d in dirs:
+        if d.startswith("measurements"):
+          yield os.path.join(root, d)
+          break
+
+
+  def instrument(self, output_dir):
```

Comment:
Can't seem to find where this method is referenced

Reply:
Referenced in junit_run.py here:https://github.com/pantsbuild/pants/blob/81993f46b2d862fc7a3a52fb51c15f3e59341736/src/python/pants/backend/jvm/tasks/junit_run.py#L257-L263